packages/tx/test/inputValue.spec.ts

```diff
@@ -0,0 +1,139 @@
+}
+
+// @returns: Array with subtypes of the BufferLike type for a given string
+function generateBufferLikeValues(value: string): BufferLike[] {
+  return [value, toBuffer(value)]
+}
+
+interface GenerateCombinationsArgs {
+  options: { [x: string]: any }
+  optionIndex?: number
+  results?: { [x: string]: any }[]
+  current?: { [x: string]: any }
+}
+
+export function generateCombinations({
+  options,
+  optionIndex = 0,
+  results = [],
+  current = {},
+}: GenerateCombinationsArgs) {
+  const allKeys = Object.keys(options)
+  const optionKey = allKeys[optionIndex]
+  const values = options[optionKey]
+
+  for (let i = 0; i < values.length; i++) {
+    current[optionKey] = values[i]
+
+    if (optionIndex + 1 < allKeys.length) {
+      generateCombinations({ options, optionIndex: optionIndex + 1, results, current })
+    } else {
+      // Clone the object
+      const res = { ...current }
+      results.push(res)
+    }
+  }
+
+  return results
+}
+
+// Deterministic pseudorandom number generator
+function mulberry32(seed: number) {
+  let t = (seed += 0x6d2b79f5)
+  t = Math.imul(t ^ (t >>> 15), t | 1)
+  t ^= t + Math.imul(t ^ (t >>> 7), t | 61)
+  return ((t ^ (t >>> 14)) >>> 0) / 4294967296
+}
+
+function getRandomSubarray<TArrayItem>(array: TArrayItem[], size: number) {
+  const shuffled = array.slice(0)
+  let seed = 1559
+  let index: number
+  let length = array.length
+  let temp: TArrayItem
+  while (length > 0) {
+    index = Math.floor((length + 1) * mulberry32(seed))
+    temp = shuffled[index]
+    shuffled[index] = shuffled[length]
+    shuffled[length] = temp
+    seed++
+    length--
+  }
+  return shuffled.slice(0, size)
+}
+
+const baseTxValues = {
+  data: generateBufferLikeValues('0x65'),
+  gasLimit: generateBNLikeValues(100000),
+  nonce: generateBNLikeValues(0),
+  to: generateAddressLikeValues('0x0000000000000000000000000000000000000000'),
+  r: generateBNLikeValues(100),
+  s: generateBNLikeValues(100),
+  value: generateBNLikeValues(10),
+}
+
+const legacyTxValues = {
+  gasPrice: generateBNLikeValues(100),
+}
+
+const accessListEip2930TxValues = {
+  chainId: generateBNLikeValues(4),
+}
+
+const eip1559TxValues = {
+  maxFeePerGas: generateBNLikeValues(100),
+  maxPriorityFeePerGas: generateBNLikeValues(50),
+}
+
+tape('[Transaction Input Values]', function (t) {
+  t.test('Legacy Transaction Values', function (st) {
+    const common = new Common({ chain: Chain.Mainnet, hardfork: Hardfork.Homestead })
+    const options = { ...baseTxValues, ...legacyTxValues, type: '0' }
+    const legacyTxData = generateCombinations({
+      options,
+    })
+    const expectedHash = Transaction.fromTxData(legacyTxData[0]).hash()
+    const randomSample = getRandomSubarray(legacyTxData, 100)
+    for (const txData of randomSample) {
+      const tx = Transaction.fromTxData(txData, { common })
+
+      st.deepEqual(tx.hash(), expectedHash), 'correct tx hash'
```

Comment:
This is a fun one (analog below): the test output string is actually outside the function call and is not taken but text output is "should be deeply equivalent". 😅 

Totally love this, that even the syntax here seems to work (what kind of syntax is that actually?) Did some tests: `'test', 'toast'` actually throws with "Left side of comma operator is unused and has no side effects." but `let t = function() {}` followed by `t(), 'test'` actually works and outputs the string. Lol.

Anyhow: maybe you can update! 🙂  Is there a (simple) way to get some of the test data into the output message too? Or is this too complicated respectively get's too verbose with the current code structure? Would be nice to identify single test cases, but also no absolute must.

Otherwise PR looks really great! 👍

Reply:
Ah! Nice catch, that syntax is really weird indeed, now fixed!

I've added the `tx.hash()` to the output message, this at least allows us to differentiate between transaction objects (e.g. the legacy case vs. the eip1559 case). I think it would be too verbose to display the tx args themselves.

Reply:
Yeah, that's a good compromise, thanks!